File: Easy/Alternate positive and negative numbers/alternate-positive-and-negative-numbers.py
#User function Template for python3

class Solution:
    def rearrange(self,arr, n):
        # code here
        # p=arr[0]
        l=[]
        r=[]
        res=[]
        for i in range(n):
            if(arr[i]>=0):
                l.append(arr[i])
            else:
                r.append(arr[i])
        i=0
        j=0
        while(i<len(l) and j<len(r)):
            res.append(l[i])
            res.append(r[j])
            i+=1
            j+=1
        while(i<len(l)):
            res.append(l[i])
            i+=1
        while(j<len(r)):
            res.append(r[j])
            j+=1
        arr[:] = res[:]
        return arr
        
        
            
        

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
	tc = int(input())
	while tc > 0:
		n = int(input())
		arr = list(map(int, input().strip().split()))
		ob = Solution()
		ob.rearrange(arr, n)
		for x in arr:
			print(x, end=" ")
		print()
		tc -= 1

# } Driver Code Ends

File: Difficulty: Easy/Alternate positive and negative numbers/alternate-positive-and-negative-numbers.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;


// } Driver Code Ends
// User function template for C++
class Solution {
  public:

    void rearrange(vector<int> &arr) {
        int n = arr.size();
        vector<int>positive;
        vector<int>negative;
        
        for(auto i:arr){
            if(i >= 0){
                positive.push_back(i);
            }
            else{
                negative.push_back(i);
            }
        }
        
        int s = 0 ;
        int e = 0 ;
        int k = 0 ;
        while(k < n && s < positive.size() && e < negative.size()){
            if(k % 2 == 0){
                arr[k] = positive[s];
                s++;
                k++;
            }
            else{
                arr[k] = negative[e];
                e++;
                k++;
            }
        }
        
        while(k < n && s < positive.size()){
            arr[k] = positive[s];
            k++;
            s++;
        }
        
        while(k < n && e < negative.size()){
            arr[k] = negative[e];
            k++;
            e++;
        }        
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    cin.ignore();
    while (t--) {
        vector<int> arr;
        string input;
        int num;

        getline(cin, input);
        stringstream s2(input);
        while (s2 >> num) {
            arr.push_back(num);
        }
        Solution ob;
        ob.rearrange(arr);
        for (int i = 0; i < arr.size(); i++) {
            cout << arr[i] << " ";
        }
        cout << "\n";
    }
    return 0;
}

// } Driver Code Ends